doc/03_reference/cs_registers.rst

```diff
@@ -309,7 +309,11 @@ Reset Value: ``0x0000_0000``
 
 Accessible in Debug Mode or M-Mode when trigger support is enabled (using the DbgTriggerEn parameter).
 
-Ibex implements a single trigger, therefore this register will always read as zero.
+Number of the currently selected trigger.
+The number of triggers is configured by the DbgHwNumLen parameter.
+
+Writing a value larger than the number of supported triggers will write a truncated number to the register.
```

Comment:
Might just be my interpretation, but truncated kind-of implies something different to what happens here. Looks like we actually just retain the old value, but could probably also set it to the supported maximum.

Reply:
True, I will have a look at the specification and just check what should actually happen.

Reply:
I think what we do now is probably ok, but it might be more useful to software to return the max rather than the current value. Then you could write, say 7 and read back 4 and know right away that there are 4 supported.

Reply:
I think the way gdb works is that the first time a breakpoint is set, it selects values until the maximum is determined.
So it might be more useful to software to return the max, but probably only gdb is the software which actually will interface with it? Only guessing, but maybe it is easier in software to determine the max by iterating than to guess a value and then guess until the max is reached?

Anyway, the spec allows some flexibility to `tselect` but looking at the description for `tdata` it would make more sense to align it and write the max value.

For reference:
current (0.13.2) debug spec
5.2.1 "Writes of values greater than or equal to the number of supported triggers may result in a different value  in  this  register  than  what  was  written.  To  verify  that  what  they  wrote  is  a  valid  index,debuggers can read back the value and check that tselect holds what they wrote."

5.2 "Most trigger functionality is optional.  All tdata registers follow write-any-read-legal semantics. If a debugger writes an unsupported configuration, the register will read back a value that is supported(which  may  simply  be  a  disabled  trigger)."